### ISSUE TITLE
Google Repository v32 breaks

### ISSUE THREAD

**Issue by whs** [zo0r/react-native-push-notification#171]
When upgrading Google Repository in Android SDK from v28 to v32, the compiled apk no longer runs:

The application will either redbox on debug build, or crash on production build

```
Fatal Exception: com.facebook.react.modules.core.JavascriptException: C++ Exception in 'nativeRequireModuleConfig': java.lang.IncompatibleClassChangeError: The method 'java.io.File android.support.v4.content.ContextCompat.getNoBackupFilesDir(android.content.Context)' was expected to be of type virtual but instead was found to be of type direct (declaration of 'com.google.android.gms.iid.zzd' appears in /data/app/redacted/base.apk), stack:
get@57:538
<unknown>@470:52
o@2:517
e@2:326
i@2:188
<unknown>@469:209
o@2:517
e@2:326
i@2:188
<unknown>@427:517
o@2:517
e@2:326
i@2:188
<unknown>@374:223
o@2:517
e@2:326
i@2:188
<unknown>@12:88
o@2:517
e@2:256
i@2:188
global code@639:9

       at com.facebook.react.modules.core.ExceptionsManagerModule.showOrThrowError(ExceptionsManagerModule.java:94)
       at com.facebook.react.modules.core.ExceptionsManagerModule.reportFatalException(ExceptionsManagerModule.java:78)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:319)
       at com.facebook.react.cxxbridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:139)
       at com.facebook.react.bridge.queue.NativeRunnable.run(NativeRunnable.java)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
       at android.os.Looper.loop(Looper.java:148)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:196)
       at java.lang.Thread.run(Thread.java:818)
```

Devices tested: Genymotion, Nexus 4, Nexus 4, Note 5 running Android 6.0

The application tested was configured correctly (it works completely fine until the repository was updated).

Dependency:

``` groovy
    compile ('com.google.android.gms:play-services-gcm:8.4.0') {
        force = true;
    }
```

It seems that this issue is [fixed](https://developers.google.com/android/guides/releases#may_2016_-_v90) in Play Services v9.0

**Comment by sunnyyants** [comment]
I face a similar issue after upgrading our project from RN 0.26 -> RN 0.32 and package 1.0.7 -> 2.0.2. We faced the crash once the app was trying to register the device token. After we change the play-services-gcm to 9.0.1, it is fixed. Hope it helps!

`  compile ('com.google.android.gms:play-services-gcm:9.0.1') {
        force = true;
    }
`

**Comment by npomfret** [comment]
I'm using RN 0.33 and gcm 9.4 and can confirm it works (on Android 5.1 and 4.1 at least):

```
compile('com.google.android.gms:play-services-gcm:9.4.0') {
        force = true;
    }

```

I think changing the docs is a good idea,

**Comment by teyou** [comment]
I'm using
- RN: 0.34.1

I was having exactly the same exception. 
Hence, I update my package to:
- Google Repository v38
- React-native-push-notification: 2.2.0
- gcm to 9.4.0

However, the only problem I've encountered now is on Android device.
No matter when the app is running on background or been killed, I could see the push notification delivered via Logcat.
The only behaviour that is different after I updated the package and changed the gcm is that
while my app receive a "noisy" (mentioned: https://github.com/zo0r/react-native-push-notification/issues/207#issuecomment-257954473) remote push notification and is running on background or been killed, by right, it should appear on notification centre but it doesn't.

Before I update, no matter app is running on background or been killed, I always see the push notification shown on the notification centre.

Anyone having this same issue?
